# DP: Proposed patch for PR rtl-optimization/49847 (cc0 targets).
gcc/

2012-02-26  Mikael Pettersson  <mikpe@it.uu.se>

	PR rtl-optimization/49847
	* cse.c (fold_rtx) <case CC0>: If prev_insn_cc0 is zero
	don't call equiv_constant on it.

--- a/gcc/cse.c
+++ b/gcc/cse.c
@@ -3194,9 +3194,14 @@ fold_rtx (rtx x, rtx insn)
 
 #ifdef HAVE_cc0
 	  case CC0:
-	    folded_arg = prev_insn_cc0;
-	    mode_arg = prev_insn_cc0_mode;
-	    const_arg = equiv_constant (folded_arg);
+	    if (!prev_insn_cc0)
+	      const_arg = 0;
+	    else
+	      {
+		folded_arg = prev_insn_cc0;
+		mode_arg = prev_insn_cc0_mode;
+		const_arg = equiv_constant (folded_arg);
+	      }
 	    break;
 #endif
 
